ethereum/rpc/namespaces/eth/api.go

```diff
@@ -538,11 +557,21 @@ func (e *PublicAPI) doCall(
 	if err != nil {
 		return nil, err
 	}
-	req := evmtypes.EthCallRequest{Args: bz}
+	req := evmtypes.EthCallRequest{Args: bz, GasCap: ethermint.DefaultRPCGasLimit}
+
+	// From ContextWithHeight: if the provided height is 0,
+	// it will return an empty context and the gRPC query will use
+	// the latest block height for querying.
```

Comment:
Not sure it the contextWithHeight is actually working.

Btw if the provided height is 0, shouldnt it uses the earliest block height instead? Maybe comment needs to be changed

Reply:
`blockNr.Int64()` will convert 0(earliest) -> 1, and negative numbers(pending/latest) -> 0.